include/vm_map.h

```diff
@@ -26,7 +27,9 @@ typedef struct vm_map {
   TAILQ_HEAD(vm_map_list, vm_map_entry) list;
   SPLAY_HEAD(vm_map_tree, vm_map_entry) tree;
   size_t nentries;
-  pmap_t *pmap;
+  pmap_t *const pmap;
```

Comment:
By explicitly marking that the pointer is const we signify that it's value never changes during a `vm_map_t` lifetime, and thus it may be safely dereferenced with no synchronization mechanisms. As C does not have structure constructors, in order to set a `const`-qualified member field it must be accessed via a non-`const` pointer - which is okay during structure initialization, since it *does not* break the semantics of "never changes during lifetime". The one-time assignment would look less "ugly" if I did not choose to perform it in a single line (compare: `pmap_t** q = (pmap_t**)&x;    *q = p;`), but still it is a common C idiom. We could wrap it in a macro: `CONST_ASSIGN(pmap_t*, x, p);` if this looks better to you. By the way, such const-evading assignment is valid in C, but not in C++, where it triggers undefined behavior.